compiler/modules/types/list.bal

```diff
@@ -270,15 +270,17 @@ function listInhabited(Context cx, FixedLengthArray members, SemType rest, Conju
         // SemType d1 = diff(s[1], t[1]);
         // return !isEmpty(cx, d1) &&  tupleInhabited(cx, [s[0], d1], neg.rest);
         // We can generalize this to tuples of arbitrary length.
-        foreach int i in 0 ..< len {
-            SemType d = diff(fixedArrayGet(members, i), listMemberAt(nt.members, nt.rest, i));
-            if !isEmpty(cx, d) {
-                FixedLengthArray s = fixedArrayShallowCopy(members);
-                fixedArraySet(s, i, d);
-                if listInhabited(cx, s, rest, neg.next) {
-                    return true;
+        if len != 0 {
```

Comment:
Are you sure you need this test?

I would just compute separately the upper limit (which might be 0) and then do the foreach up to that limit (which will do nothing if the upper limit is 0).

Reply:
No, we don't need it (https://github.com/ballerina-platform/nballerina/pull/827/files#r778093687) I just don't have a name for the variable.
Would something like `distinctLen` work?